### DISCUSSION TITLE
How do I know what API or model was used by default for any generated output?

### DISCUSSION THREAD

**Discussion by itsRevela** [danielmiessler/Fabric#381] [Q&A]
- How do I know what model a pattern used by default?
- If I do use a pattern, are there logs or debug messages available that describe what model is being used, what's going on in the backend, etc?

**Comment by xssdoctor** [comment]
It doesn't work like that. You have a default model that all patterns use. You can also specify --model that chooses a model. Any pattern can be used with any model

**Reply by itsRevela** [reply]
I think you may have misunderstood:
- I setup Fabric with 3 different api keys (claude/openai/googleyoutube). 
- I then ran a pattern without specifying a model. 
- I already know that models are not specific to a pattern.

**How do I know what API or model was used by default for the generated output?**
